app/models/queue_item.rb

```diff
@@ -2,15 +2,33 @@ class QueueItem < ActiveRecord::Base
   belongs_to :video
   belongs_to :user
   validates_presence_of :video, :user
+  validates_numericality_of :order, only_integer: true
   delegate :title, to: :video, prefix: :video
   delegate :name, to: :category, prefix: :category
 
   def rating
-    video.reviews.first.rating
+    review.rating if review
+  end
+
+  def rating=(rating_num)
+    if review
+      review.update_attribute(:rating, rating_num) 
+    else
+      new_review = Review.new(
+        creator: user, 
+        rating: rating_num,
+        video_id: video.id
```

Comment:
You could use association `video` and then have `video: video` instead of having appended 'id'. Its only for beauty and more succinct code, nothing required.